Upgrade to GraalVM for JDK 24

Akirathan opened the pull request:
Closes #12452

Upgrade GraalVM from JDK 21.0.2 to **JDK 24.0.1**.

Download JDK 24.0.1 from https://github.com/graalvm/graalvm-ce-builds/releases/tag/jdk-24.0.1. For users of SDKMan: `sdk install java 24.0.1-graalce`

### Native image
- Using newly introduced `-Os` instead of `-O3` for production build. See https://github.com/enso-org/enso/pull/12855/commits/53ff5a6a0dd980f31d460776f6fce03af7256573
  - Size of the production NI decreased from 454 MB to 431 MB. See https://github.com/enso-org/enso/pull/12855#issuecomment-2815611629
- Build times are generally much faster. See https://github.com/enso-org/enso/pull/12855#issuecomment-2812552448

### Relevant changelogs
- https://github.com/oracle/graal/blob/master/truffle/CHANGELOG.md
- https://github.com/oracle/graal/blob/master/sdk/CHANGELOG.md
  - Native access has to be enabled with `--enable-native-access=org.graalvm.truffle`.
- https://github.com/oracle/graal/blob/master/compiler/CHANGELOG.md
- https://github.com/oracle/graal/blob/master/substratevm/CHANGELOG.md
  - (GR-52844) Add `-Os`, a new optimization mode to configure the optimizer in a way to get the smallest code size.
  - (GR-54241) Streamline Native Image reachability metadata into a single `reachability-metadata.json`. See [Reachability metadata docs](https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/ReachabilityMetadata.md).

Terminal deprecation of most methods from `sun.misc.Unsafe`: https://openjdk.org/jeps/498

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.
- [x] If meaningful changes were made to logic or tests affecting Enso Cloud integration in the libraries, 
      or the Snowflake database integration, a run of the [Extra Tests](https://github.com/enso-org/enso/actions/workflows/extra-nightly-tests.yml) has been scheduled.
  - If applicable, it is suggested to paste a link to a successful run of the Extra Tests.

Commit:
Rename jdk.internal.vm.compiler to jdk.graal.compiler

Commit:
Resolve warning when starting sbt

Commit:
Update Java version and Graal version in build.sbt

Commit:
jdk.internal.vm.compiler does not exist

Commit:
Fix class file version of task-progress-notifications

Commit:
Fix graphio package name

Commit:
Update changelog

Commit:
SemVer parsing deals with unspecified minor and patch versions

Commit:
graalVersion in build.sbt does not have to be semver

Commit:
downloaded graalvm dir does not have to contain the whole version

Commit:
enable native access for org.graalvm.truffle module in cmd line launch scripts

Commit:
Allow memory access in sun.misc.Unsafe in cmd line launch scripts.

HotspotTruffleRuntime (24.2.0) uses terminally deprecated methods from sun.misc.Unsafe. This is to ensure there is no warning printed.

Akirathan commented:
HotspotTruffleRuntime 24.2.0 still uses terminally deprecated memory access methods from `sun.misc.Unsafe`. Running enso with:
```
env JAVA_OPTS='--sun-misc-unsafe-memory-access=deny' enso --run tmp.enso
```
fails with `UnsupportedOperationException`:
```
Exception in thread "main" java.lang.InternalError: java.lang.UnsupportedOperationException: objectFieldOffset
	at org.graalvm.truffle.runtime/com.oracle.truffle.runtime.hotspot.HotSpotTruffleRuntime.<clinit>(HotSpotTruffleRuntime.java:112)
	at org.graalvm.truffle.runtime/com.oracle.truffle.runtime.hotspot.HotSpotThreadLocalHandshake.<clinit>(HotSpotThreadLocalHandshake.java:55)
	at org.graalvm.truffle.runtime/com.oracle.truffle.runtime.hotspot.HotSpotTruffleRuntimeAccess.createRuntime(HotSpotTruffleRuntimeAccess.java:217)
	at org.graalvm.truffle.runtime/com.oracle.truffle.runtime.hotspot.HotSpotTruffleRuntimeAccess.getRuntime(HotSpotTruffleRuntimeAccess.java:76)
	at org.graalvm.truffle/com.oracle.truffle.api.Truffle.createRuntime(Truffle.java:145)
	at org.graalvm.truffle/com.oracle.truffle.api.Truffle$1.run(Truffle.java:176)
	at org.graalvm.truffle/com.oracle.truffle.api.Truffle$1.run(Truffle.java:174)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:74)
	at org.graalvm.truffle/com.oracle.truffle.api.Truffle.initRuntime(Truffle.java:174)
	at org.graalvm.truffle/com.oracle.truffle.api.Truffle.<clinit>(Truffle.java:63)
	at org.graalvm.truffle/com.oracle.truffle.api.impl.Accessor.getTVMCI(Accessor.java:1643)
	at org.graalvm.truffle/com.oracle.truffle.api.impl.Accessor$Constants.<clinit>(Accessor.java:1497)
	at org.graalvm.truffle/com.oracle.truffle.api.impl.Accessor.engineSupport(Accessor.java:1564)
	at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryAccessor.engineAccessor(LibraryAccessor.java:57)
	at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryFactory.loadExternalDefaultProviders(LibraryFactory.java:439)
	at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryFactory.getExternalDefaultProviders(LibraryFactory.java:430)
	at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryFactory.initDefaultExports(LibraryFactory.java:213)
	at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryFactory.<init>(LibraryFactory.java:208)
	at org.graalvm.truffle/com.oracle.truffle.api.library.DynamicDispatchLibraryGen.<init>(DynamicDispatchLibraryGen.java:33)
	at org.graalvm.truffle/com.oracle.truffle.api.library.DynamicDispatchLibraryGen.<clinit>(DynamicDispatchLibraryGen.java:25)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:543)
	at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryFactory.loadGeneratedClass(LibraryFactory.java:775)
	at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryFactory.resolveImpl(LibraryFactory.java:756)
	at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryFactory.resolve(LibraryFactory.java:749)
	at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryFactory.<init>(LibraryFactory.java:202)
	at org.graalvm.truffle/com.oracle.truffle.api.interop.InteropLibraryGen.<init>(InteropLibraryGen.java:180)
	at org.graalvm.truffle/com.oracle.truffle.api.interop.InteropLibraryGen.<clinit>(InteropLibraryGen.java:171)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:543)
	at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryFactory.loadGeneratedClass(LibraryFactory.java:775)
	at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryFactory.resolveImpl(LibraryFactory.java:756)
	at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryFactory.resolve(LibraryFactory.java:749)
	at org.graalvm.truffle/com.oracle.truffle.api.interop.InteropLibrary.<clinit>(InteropLibrary.java:2972)
	at org.graalvm.truffle/com.oracle.truffle.polyglot.PolyglotValueDispatch.<clinit>(PolyglotValueDispatch.java:168)
	at org.graalvm.truffle/com.oracle.truffle.polyglot.PolyglotImpl.initialize(PolyglotImpl.java:208)
	at org.graalvm.polyglot/org.graalvm.polyglot.Engine.loadAndValidateProviders(Engine.java:1811)
	at org.graalvm.polyglot/org.graalvm.polyglot.Engine$1.run(Engine.java:1837)
	at org.graalvm.polyglot/org.graalvm.polyglot.Engine$1.run(Engine.java:1832)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:74)
	at org.graalvm.polyglot/org.graalvm.polyglot.Engine.initEngineImpl(Engine.java:1832)
	at org.graalvm.polyglot/org.graalvm.polyglot.Engine$ImplHolder.<clinit>(Engine.java:177)
	at org.graalvm.polyglot/org.graalvm.polyglot.Engine.getImpl(Engine.java:446)
	at org.graalvm.polyglot/org.graalvm.polyglot.Engine$Builder.build(Engine.java:755)
	at org.graalvm.polyglot/org.graalvm.polyglot.Context$Builder.build(Context.java:1975)
	at org.enso.engine.common/org.enso.common.ContextFactory.build(ContextFactory.java:264)
	at org.enso.runner/org.enso.runner.Main.handleRun(Main.java:782)
	at org.enso.runner/org.enso.runner.Main.mainEntry(Main.java:1166)
	at org.enso.runner/org.enso.runner.Main.lambda$launch$25(Main.java:1660)
	at org.enso.runner/org.enso.runner.Main.withProfiling(Main.java:1274)
	at org.enso.runner/org.enso.runner.Main.launch(Main.java:1656)
	at org.enso.runner/org.enso.runner.Main.launch(Main.java:1526)
	at org.enso.runner/org.enso.runner.Main.main(Main.java:1075)
Caused by: java.lang.UnsupportedOperationException: objectFieldOffset
	at jdk.unsupported/sun.misc.Unsafe.beforeMemoryAccessSlow(Unsafe.java:1822)
	at jdk.unsupported/sun.misc.Unsafe.beforeMemoryAccess(Unsafe.java:1787)
	at jdk.unsupported/sun.misc.Unsafe.objectFieldOffset(Unsafe.java:905)
	at org.graalvm.truffle.runtime/com.oracle.truffle.runtime.hotspot.HotSpotTruffleRuntime.getObjectFieldOffset(HotSpotTruffleRuntime.java:670)
	at org.graalvm.truffle.runtime/com.oracle.truffle.runtime.hotspot.HotSpotTruffleRuntime.<clinit>(HotSpotTruffleRuntime.java:110)
	... 52 more
```

I am adding `--sun-misc-unsafe-memory-access=allow` to cmdline options in https://github.com/enso-org/enso/pull/12855/commits/873115b90a796a764e0086c6473ff10baaea1c43 to avoid unnecessary warnings. See https://openjdk.org/jeps/498

Commit:
Allow unsafe memory access for sbt

Commit:
Frgaal has access to jdk.unsupported module

Commit:
Enable native access globally in sbt

Commit:
Fix package name in jdk.graal.compiler graphio

Commit:
Merge branch 'develop' into wip/akirathan/12452-upgrade-graal-24-2-0

Commit:
Allow concurrent reachability handlers run in engine-runner NI

Commit:
Add necessary cmd line opts when running with --jvm

Commit:
DiagnosticFormatter does not access EnsoContext.

Caused AssertionError: No polyglot context entered in tests.

Commit:
Fix graphio package name in IGV dumper

JaroslavTulach commented on CHANGELOG.md:
```diff
@@ -189,6 +189,7 @@
 - [Registering a value as multiple managed resources is now an error.][12395]
 - [An operator block now applies to the whole preceding expression][12505],
   rather than the last term on the line.
+- [Upgrade GraalVM to from JDK 21 to JDK 24][12855]
```
```suggestion
- [Upgrade GraalVM from JDK 21 to JDK 24][12855]
```

JaroslavTulach commented on engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/DiagnosticFormatterTest.java:
```diff
@@ -15,18 +15,29 @@
 
+  @BeforeClass
+  public static void before() {
+    System.setProperty("NO_COLOR", "true");
```
Setting and clearing a system property in `@BeforeClass` & co. seems like the last _level of rescue_. This is just a single test that runs in the same JVM. What if some other test initializes the system that reads `NO_COLOR` before?

JaroslavTulach commented on engine/runtime/src/main/scala/org/enso/interpreter/runtime/util/DiagnosticFormatter.scala:
```diff
@@ -155,7 +154,37 @@ class DiagnosticFormatter(
+    * Checking also system props, so that we can conveniently test this.
+    */
+  private def isColorTerminalOutput: Boolean = {
+    if (
```
Having a `RuntimeOptions` option to influence `EnsoContext.isColorTerminalOutput` would seem less hackish to me.

Akirathan replied:
As explained in a6ccc00284e940d0371af2dc41d4dcf9433b4dd9, I wanted `DiagnosticFormatter` not to rely on `EnsoContext`. There were many AssertionErrors "No polyglot context entered" in tests. I realize however, that the way I am testing it in [DiagnosticFormatterTest](https://github.com/enso-org/enso/blob/a6ccc00284e940d0371af2dc41d4dcf9433b4dd9/engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/DiagnosticFormatterTest.java) by explicitly setting and resetting a system property is less than optimal. If you are still convinced that having `isColorTerminalOutput` in `EnsoContext` is better idea, I can revert it and fix the previously failing tests.

JaroslavTulach replied:
FYI: https://github.com/enso-org/enso/commit/a6ccc00284e940d0371af2dc41d4dcf9433b4dd9#r155527825

The formatter is created at
```
        at org.enso.runtime/org.enso.interpreter.runtime.util.DiagnosticFormatter.<init>(DiagnosticFormatter.scala:22)
        at org.enso.runtime/org.enso.interpreter.runtime.TruffleCompilerContext.formatDiagnostic(TruffleCompilerContext.java:314)
        at org.enso.runtime/org.enso.interpreter.runtime.TruffleCompilerContext.formatDiagnostic(TruffleCompilerContext.java:58)
        at org.enso.runtime.compiler/org.enso.compiler.Compiler.$anonfun$reportDiagnostics$2(Compiler.scala:1121)
        at scala.library@2.13.15/scala.collection.immutable.List.flatMap(List.scala:294)
        at org.enso.runtime.compiler/org.enso.compiler.Compiler.reportDiagnostics(Compiler.scala:1119)
```
and `TruffleCompilerContext` has direct reference to `EnsoContext`.

Akirathan replied:
Good point. Fixed in 1ab431c051ab415897daa3f0311409222ba002a6

JaroslavTulach reviewed (approved):
Nice progress.

Akirathan commented:
# Native image size increased

Size of `engine-runner` NI increased, when compared to previous build on JDK21. Both on CI and locally. The time to build NI is significantly reduced on CI.

The following table shows that there are more transitively reachable methods and fields on JDK24, therefore, the overall size of the executable is bigger:

| URL or local command | Size | Reachable fields | Reachable methods | Time |
|--------|--------|--------|--------|--------|
| [JDK 21 CI develop](https://github.com/enso-org/enso/actions/runs/14469147478/job/40578473562#step:10:2325) | 463.53 MB | 119,505 | 453,247 | 7:43 |
| [JDK 24 CI this PR](https://github.com/enso-org/enso/actions/runs/14491001087/job/40647251535?pr=12855#step:10:2437) | **514.91 MB** | 112,385 | 464,105 | **4:06** |
| (JDK 21 develop) `env ENSO_LAUNCHER=native,-ls,fast,test sbt buildEngineDistribution` | 403.07 MB | 97,421 | 378,904 | 4:47 |
| (JDK 24 on this PR) `env ENSO_LAUNCHER=native,-ls,fast,test sbt buildEngineDistribution` | 448.36 MB | 100,305 | 388,940 | 3:51 |

With the newly introduced `-Os` option, the local native image gets down from 448.36 MB to 402.48 MB. **Let's use the `-Os` option for the production instead of `-O3`**.

Note: native-image doesn't seem to be able to pick both `-O3` and `-Os`. It is able to apply optimizations from only one of these options.

Commit:
Update native image docs

Commit:
Use -Os option in production

Commit:
Revert "DiagnosticFormatter does not access EnsoContext."

This reverts commit a6ccc00284e940d0371af2dc41d4dcf9433b4dd9.

Commit:
DiagnosticFormatter does not depend on EnsoContext

Commit:
Remove the redundant $node expression in @Bind.

Truffle DSL now generates warning for these.

Commit:
Update CHANGELOG.md

Co-authored-by: Jaroslav Tulach <jaroslav.tulach@enso.org>

Commit:
[WIP] NI build is verbose

Commit:
Append global java opts for manifest regeneration

Commit:
Initialize PrintStream with autoflush.

This may fix the verbose output in stdlib tests.

Commit:
Revert "[WIP] NI build is verbose"

This reverts commit 30c45a6147c73060cca19c36236376bd8d8a6dc0.

Akirathan commented:
Manually scheduled nightly release at https://github.com/enso-org/enso/actions/runs/14534416939 to see the production build of NI.

NI build https://github.com/enso-org/enso/actions/runs/14534416939/job/40780059725#step:10:2195 for production uses `-Os` and the resulting size is **431.55 MB**.

For comparison, the latest Nightly release on develop produces NI of **454.19 MB** at https://github.com/enso-org/enso/actions/runs/14528493698/job/40764266580#step:10:2259

<blockquote><img src="https://opengraph.githubassets.com/198817caeecae1c187ffcdae89f7d3f153cf4400a691401c15ef9047e8470c19/enso-org/enso" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">Nightly Release · enso-org/enso@1c9d93f</a></strong></div><div>Enso Analytics is a self-service data prep and analysis platform designed for data teams. - Nightly Release · enso-org/enso@1c9d93f</div></blockquote>
<blockquote><img src="https://opengraph.githubassets.com/198817caeecae1c187ffcdae89f7d3f153cf4400a691401c15ef9047e8470c19/enso-org/enso" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">Nightly Release · enso-org/enso@1c9d93f</a></strong></div><div>Enso Analytics is a self-service data prep and analysis platform designed for data teams. - Nightly Release · enso-org/enso@1c9d93f</div></blockquote>

Akirathan commented:
The output of Standard Library Tests (for example look [here](https://github.com/enso-org/enso/actions/runs/14534362676/job/40779881737?pr=12855#step:10:3796)) is off - all the progress output created by [Test_Reporter](https://github.com/enso-org/enso/blob/261939979906db0da256012bc1f7a7541c72dee0/distribution/lib/Standard/Test/0.0.0-dev/src/Test_Reporter.enso#L231) that should be cleared from the console by https://github.com/enso-org/enso/blob/261939979906db0da256012bc1f7a7541c72dee0/distribution/lib/Standard/Test/0.0.0-dev/src/Test_Reporter.enso#L246-L249 does not work and instead prints a blank newline.

This results in the job having very big output and loading it via browser takes ages. I cannot reproduce this locally. Looking into JDK22-24 changelogs does not give me any clue what could have changed. Trying to fix this behavior by specifying `autoflush` when initializing `PrintStream` in e5841495fe55ec3c17bc7f968ea028ddfcacaf94 also did not help.

### EDIT:
The problem was probably in [Test_Reporter.is_terminal_interactive](https://github.com/enso-org/enso/blob/261939979906db0da256012bc1f7a7541c72dee0/distribution/lib/Standard/Test/0.0.0-dev/src/Test_Reporter.enso#L278-L281). According to [release notes of JDK22](https://www.oracle.com/java/technologies/javase/22-relnote-issues.html#JDK-8308591) the behavior of `java.lang.System.console` changed, and it now returns non-null value even if there is no terminal connected to stdout. Pushed another fix in https://github.com/enso-org/enso/pull/12855/commits/75404eda8cf3be14767d6c8f8ae0fcc1560d949e

Commit:
Revert "Initialize PrintStream with autoflush."

This reverts commit e5841495fe55ec3c17bc7f968ea028ddfcacaf94.

Commit:
Fix Test_Reporter.is_terminal_interactive.

https://www.oracle.com/java/technologies/javase/22-relnote-issues.html#JDK-8308591

Commit:
Do not use deprecated methods in HttpServer

Commit:
Ignore value of Http2-Settings header.

Value of this header was changed in JDK24 - https://github.com/openjdk/jdk/commit/03bc6b359fc5ff4fa2b569723903cdf1c9c5775b#diff-e0ece1ea7769def2838cc95a5a94f324392ebe57f34ca6cf5bfad3ddccfac8d4

Commit:
Abrupt stream close causes different IOException

Akirathan commented:
[Failing](https://github.com/enso-org/enso/actions/runs/14537506691/job/40788761810?pr=12855#step:10:1858) HTTP_Spec tests in stdlib tests should be fixed by https://github.com/enso-org/enso/pull/12855/commits/83e2a44953a9364a9d1940a8296d107acb626313 and https://github.com/enso-org/enso/pull/12855/commits/6c795c27df773035398215d6b012a0110a869287. There were few changes in JDK's `java.net.HttpClient` and `com.sun.net.httpserver.HttpServer`.

Commit:
Use GraalVM 24 in enso4igv job

Commit:
Use GraalVM 24 in formatting job

Commit:
Use GraalVM 24.0.1

Released just 3 days ago.

Commit:
Revert "downloaded graalvm dir does not have to contain the whole version"

This reverts commit 57efe37c2e983ba553a92cba0462f568d22da462.

Commit:
Revert "graalVersion in build.sbt does not have to be semver"

This reverts commit 45924017d5e9d1057b9c91b0fec7b6f8d50d0e99.

Commit:
Revert "SemVer parsing deals with unspecified minor and patch versions"

This reverts commit 0cfa592eeb700a48439f689c393a0e03c6c78830.

Akirathan commented:
Another nightly release scheduled at https://github.com/enso-org/enso/actions/runs/14571498604

<blockquote><img src="https://opengraph.githubassets.com/4959fd080b270935e1bcc937e0fc5669eeebd926742357decb765ea6a6bfd2f8/enso-org/enso" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">Nightly Release · enso-org/enso@28357d6</a></strong></div><div>Enso Analytics is a self-service data prep and analysis platform designed for data teams. - Nightly Release · enso-org/enso@28357d6</div></blockquote>

Commit:
Remove outdated -XDignore.symbol.file option

Commit:
enso4igv target is 17

Akirathan commented:
[classfile confusion](https://github.com/enso-org/enso/actions/runs/14538801821/job/40792636077?pr=12855) in enso4igv can be fixed by removing `-XDignore.symbol.file` compiler option. Done in https://github.com/enso-org/enso/pull/12855/commits/ccb709c749b82bccb844354621059dc1bfd5a56f. When doing that, the compilation was failing with "Cannot access record". That was fixed by bumping the target from 1.8 to 17 in https://github.com/enso-org/enso/pull/12855/commits/8644e35bb77bba2c2e43c469707ab0d63d1ab464. Is that an acceptable target @JaroslavTulach?

Commit:
Remove deprecated URL constructor

Commit:
fmt

JaroslavTulach commented on docs/infrastructure/native-image.md:
```diff
@@ -316,3 +317,24 @@ $ ENSO_JAVA=espresso ./built-distribution/enso-engine-*/enso-*/bin/enso --run he
+### Helpful cmdline options
+
+- Since
+  [GraalVM JDK 24](https://github.com/graalvm/graalvm-ce-builds/releases/tag/jdk-24.0.0),
```
```suggestion
  [GraalVM JDK 24](https://github.com/graalvm/graalvm-ce-builds/releases/tag/jdk-24.0.1),
```

JaroslavTulach commented on tools/enso4igv/pom.xml:
```diff
@@ -80,10 +80,9 @@
                     <compilerId>frgaal</compilerId>
                     <source>21</source>
-                    <target>1.8</target>
+                    <target>17</target>
```
We don't control the JDK VSCode is running on. Hence, I'd suggest to support still JDK 11.  If not continue with `1.8` - I don't understand why there should be a need to upgrade to 17?

Akirathan replied:
Because it doesn't work for target 11, neither 1.8. See my [comment addressed to you](https://github.com/enso-org/enso/pull/12855#issuecomment-2818374620). Any suggestions how to make it work are welcome.

JaroslavTulach replied:
Let's [leave the resolution](https://github.com/enso-org/enso/pull/12855#issuecomment-2820048420) for later.

JaroslavTulach commented:
>Is that an acceptable target @JaroslavTulach?

I don't like it and I don't think this should be a final state. But it is not really related to the awaited upgrade of GraalVM. Let's finish the upgrade and then let me revert these `enso4igv` changes.

JaroslavTulach commented:
> [classfile confusion](https://github.com/enso-org/enso/actions/runs/14538801821/job/40792636077?pr=12855) in enso4igv can be fixed by removing `-XDignore.symbol.file` compiler option. 

- JDK is "changing" its `.class` format with every new version
- by increasing the "version number" in the `.class` 
- as such older tools (Groovy, Gradle) usually have problems reading the old format
- and have problems running on latest JDK

- [Frgaal ](http://frgaal.org) carries its own copy of JDK classes signatures and use those
- [Frgaal ](http://frgaal.org) is shielded against incrementing the JDK classes numbers
- unless it is **told to read the JDK classes** by  `-XDignore.symbol.file` compiler option. 

> Done in [ccb709c](https://github.com/enso-org/enso/commit/ccb709c749b82bccb844354621059dc1bfd5a56f). When doing that, the compilation was failing with "Cannot access record". 

When I remove  `-XDignore.symbol.file` compiler option. I see:
```
tools/enso4igv/src/main/java/org/enso/tools/enso4igv/EnsoActionProvider.java:[3,26] error: package com.sun.jdi.connect does not exist
```
e.g. the primary problem is access `com.sun.jdi.connect` package - a package not part of JDK API and as such not carried by Frgaal compiler itself. 

> That was fixed by bumping the target from 1.8 to 17 in [8644e35](https://github.com/enso-org/enso/commit/8644e35bb77bba2c2e43c469707ab0d63d1ab464). Is that an acceptable target @JaroslavTulach?

Good enough for this PR. Otherwise unfortunate.

hubertp commented on distribution/lib/Standard/Test/0.0.0-dev/src/Test_Reporter.enso:
```diff
@@ -278,4 +278,5 @@ type Command_Line_Progress_Reporter
 is_terminal_interactive -> Boolean =
-    Java_System.console != Nothing
+    console = Java_System.console
+    console != Nothing && console.isTerminal
```
this works in NI?

Akirathan replied:
You are right, it does not. Pushing the fix in https://github.com/enso-org/enso/pull/12930. But it works on CI, since on CI, it seems that `console != Nothing` evaluates to `False`, and so `console.isTerminal` is not even called.

Commit:
Add scala.Symbol to reachability metadata

GregoryTravis reviewed (approved):
Libs review
